apps/settings/js/panels/operator_settings/models/network_type_manager.js

```diff
@@ -0,0 +1,141 @@
+/* global getSupportedNetworkInfo */
+/**
+ * NetworkTypeManager provides a supported network type list of a mobile
+ * connection. It wraps setPreferredNetworkType and is responsible for updating
+ * the settings field when setting preferred network types.
+ *
+ * @module panels/operator_settings/models/network_type_manager
+ */
+define(function(require) {
+  'use strict';
+
+  var Module = require('modules/base/module');
+  var Observable = require('modules/mvvm/observable');
+  var ObservableArray = require('modules/mvvm/observable_array');
+  var SettingsHelper = require('shared/settings_helper');
+
+  const SETTINGS_KEY = 'ril.radio.preferredNetworkType';
+
+  /**
+   * @class NetworkTypeManager
+   * @requires module:modules/base/module
+   * @requires module:modules/mvvm/observable
+   * @requires module:modules/mvvm/observable_array
+   * @requires module:shared/settings_helper
+   * @params {MozMobileConnection} conn
+   * @returns {NetworkTypeManager}
+   */
+  var NetworkTypeManager = Module.create(function NetworkTypeManager(conn) {
+    if (!conn.setPreferredNetworkType) {
+      this.throw(`mobile connection does not support settings preferred network
+        type`);
+    }
+    this.super(Observable).call(this);
+
+    this._conn = conn;
+    this._serviceId = [].indexOf.call(navigator.mozMobileConnections, conn);
+    this._networkTypes = ObservableArray([]);
+    this._supportedNetworkInfo = null;
+    this._settingsHelper = SettingsHelper(SETTINGS_KEY);
+
+    this._init(conn);
+  }).extend(Observable);
+
+  /**
+   * Available network types.
+   *
+   * @access public
+   * @readonly
+   * @memberOf NetworkTypeManager.prototype
+   * @type {ObservableArray.<String>}
+   */
+  Object.defineProperty(NetworkTypeManager.prototype, 'networkTypes', {
+    enumerable: true,
+    get: function() {
+      return this._networkTypes;
+    }
+  });
+
+  /**
+   * An observable property indicating the preferred network type.
+   *
+   * @access public
+   * @readonly
+   * @memberOf NetworkTypeManager.prototype
+   * @type {String}
+   */
+  Observable.defineObservableProperty(NetworkTypeManager.prototype,
+    'preferredNetworkType', {
+      readonly: true,
+      value: null
+  });
+
+  /**
+   * Initialize the supported network type list.
+   *
+   * @access private
+   * @memberOf NetworkTypeManager.prototype
+   * @params {MozMobileConnection} conn
+   * @returns {Promise}
+   */
+  NetworkTypeManager.prototype._init = function(conn) {
```

Comment:
You already used `this._conn` to do related works, let's remove this param here.